test_runner/logic_test_util.py

```diff
@@ -67,9 +71,30 @@ def RunLogicTestOnSim(sim_id,
   developer_dir = os.environ.get('DEVELOPER_DIR')
   if developer_dir:
     simctl_env_vars['DEVELOPER_DIR'] = developer_dir
+
+  xctest_tool = xcode_info_util.GetXctestToolPath(ios_constants.SDK.IPHONESIMULATOR)
+
+  test_bundle_name = os.path.splitext(os.path.basename(test_bundle_path))[0]
+  test_executable = os.path.join(test_bundle_path, test_bundle_name)
+  test_archs = bundle_util.GetFileArchTypes(test_executable)
+  
+  # if a logic bundle is built w/ x86 on Apple silicon, it won't be able to launch on the ARM64 sim; rework the xctest to fix this
+  if ios_constants.ARCH.X86_64 in test_archs and len(test_archs) == 1:
+    replacement_xctest_tool = os.path.join(tempfile.mkdtemp(), "xctest")
```

Comment:
```suggestion
    replacement_xctest_tool = os.path.join(os.path.dirname(xctest_tool), "xctest_x86_64")
```
this fixes swift logic test issue with xcode 13.3 and ios 15.4 as @albertdai suspected here https://github.com/google/xctestrunner/pull/40#issuecomment-1101732521

Reply:
this could fail depending on how the user installed xcode, as it won't have permissions in some cases